docs/config.md

```diff
@@ -782,6 +808,78 @@ The base64 encoded protobuf describing (and containing) the node's private key.
 
 Type: `string` (base64 encoded)
 
+## `Internal`
+
+### `Internal.Bitswap`
+
+InternalBitswap contains knobs for tuning bitswap resource utilization.
+The knobs (below) document how their value should related to each other.
+Whether their values should be raised or lowered should be determined
+based on the metrics `ipfs_bitswap_active_tasks`, `ipfs_bitswap_pending_tasks`,
+`ipfs_bitswap_pending_block_tasks` and `ipfs_bitswap_active_block_tasks`
+reported by bitswap.
+
+These metrics can be accessed as the prometheus endpoint `API_ADDR/debug/metrics/prometheus`.
+
+The value of `ipfs_bitswap_active_tasks` is capped by `EngineTaskWorkerCount`.
+
+The value of `ipfs_bitswap_pending_tasks` is generally capped by the knobs below,
+however its exact maximum value is hard to predict as it depends on task sizes
+as well as number of requesting peers. However, as a rule of thumb,
+during healthy operation this value should oscillate around a "typical" low value
+(without hitting a plateau continuously).
+
+If `ipfs_bitswap_pending_tasks` grows and eventually reaches a plateau,
+while at the same time `ipfs_bitswap_active_tasks` is at its maximum,
+the node has reached its resource limits and requests are being dropped (or not serviced).
+Raising resource limits (using the knobs below) could help, assuming the hardware can support the new limits.
+
+The value of `ipfs_bitswap_active_block_tasks` is capped by `EngineBlockstoreWorkerCount`.
+
+The value of `ipfs_bitswap_pending_block_tasks` is indirectly capped by `ipfs_bitswap_active_tasks`, but cannot be predicted
+as it depends on the number of blocks involved in a peer task which can vary.
+
+If the value of `ipfs_bitswap_pending_block_tasks` is observed to grow,
+while `ipfs_bitswap_active_block_tasks` is at its maximum, there is indication that the number of
+available block tasks is creating a bottleneck (either due to high-latency block operations,
+or due to high number of block operations per bitswap peer task).
+In such cases, try increasing the `EngineBlockstoreWorkerCount`.
+If this adjustment still does not increase the throuput of the node, there might
+be hardware limitations like I/O or CPU.
+
+#### `Internal.Bitswap.TaskWorkerCount`
+
+Number of threads sending outgoing messages.
+Used to thottle the number of concurrent send operations.
+
+Type: `integer` (thread count, 0 means default which is 8)
```

Comment:
nit: can we switch those to custom type where `null` / missing indicates the default, as suggested in https://github.com/ipfs/go-ipfs-config/pull/139#discussion_r683504132